bee-ledger/src/workers/consensus/white_flag.rs

```diff
@@ -0,0 +1,298 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    types::{BalanceDiffs, ConsumedOutput, CreatedOutput},
+    workers::{
+        consensus::{merkle_hasher::MerkleHasher, metadata::WhiteFlagMetadata},
+        error::Error,
+        storage::{self, StorageBackend},
+    },
+};
+
+use bee_message::{
+    address::Address,
+    input::Input,
+    output::{dust_outputs_max, Output, OutputId, DUST_THRESHOLD},
+    payload::{
+        transaction::{Essence, RegularEssence, TransactionId, TransactionPayload},
+        Payload,
+    },
+    unlock::{UnlockBlock, UnlockBlocks},
+    Message, MessageId,
+};
+use bee_tangle::{ConflictReason, MsTangle};
+
+use crypto::hashes::blake2b::Blake2b256;
+
+use std::collections::{HashMap, HashSet};
+
+fn verify_signature(address: &Address, unlock_blocks: &UnlockBlocks, index: usize, essence_hash: &[u8; 32]) -> bool {
+    match unlock_blocks.get(index) {
```

Comment:
I think clippy will scream at you for this one

Reply:
It doesn't, why should it ?

Reply:
You're matching on a single variant